error.go

```diff
@@ -95,6 +95,15 @@ func IsPullChartFailedError(err error) bool {
 	return microerror.Cause(err) == pullChartFailedError
 }
 
+var pullChartNotFoundError = &microerror.Error{
+	Kind: "pullChartNotFoundError",
+}
```

Comment:
Currently we are returning the same error for host not found and 404s for missing tarballs.

I think this is wrong because host not found is due to an incorrect appcatalog CR. But we can get tarball 404s due to timing issues.